lib/CL/poclcc_binary.c

```diff
@@ -0,0 +1,443 @@
+#include "poclcc_binary.h"
+
+void poclcc_free(poclcc_global *binary_format){
+  if (binary_format != NULL)
+    if (binary_format->devices != NULL)
+      {
+        int i;
+        for (i=0; i<binary_format->num_devices; i++)
+          poclcc_device_free(&(binary_format->devices[i]));
+        POCL_MEM_FREE(binary_format->devices);
+      }
+}
+
+void poclcc_device_free(poclcc_device *device){
```

Comment:
Then this also should be pocl_binary_device? poclcc is the binary for building kernels to pocl_binaries (like nvcc in NVIDIA).